src/Analyzers/CSharp/CodeFixes/UseImplicitOrExplicitType/UseExplicitTypeCodeFixProvider.cs

```diff
@@ -63,63 +62,123 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
             Document document, SyntaxEditor editor,
             SyntaxNode node, CancellationToken cancellationToken)
         {
-            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
             var declarationContext = node.Parent;
 
-            TypeSyntax typeSyntax = null;
-            ParenthesizedVariableDesignationSyntax parensDesignation = null;
             if (declarationContext is RefTypeSyntax refType)
             {
                 declarationContext = declarationContext.Parent;
             }
 
             if (declarationContext is VariableDeclarationSyntax varDecl)
             {
-                typeSyntax = varDecl.Type;
+                await HandleVariableDeclarationAsync(document, editor, varDecl, cancellationToken).ConfigureAwait(false);
             }
             else if (declarationContext is ForEachStatementSyntax forEach)
             {
-                typeSyntax = forEach.Type;
+                await HandleForEachStatementAsync(document, editor, forEach, cancellationToken).ConfigureAwait(false);
             }
             else if (declarationContext is DeclarationExpressionSyntax declarationExpression)
             {
-                typeSyntax = declarationExpression.Type;
-                if (declarationExpression.Designation.IsKind(SyntaxKind.ParenthesizedVariableDesignation, out ParenthesizedVariableDesignationSyntax variableDesignation))
-                {
-                    parensDesignation = variableDesignation;
-                }
+                await HandleDeclarationExpressionAsync(document, editor, declarationExpression, cancellationToken).ConfigureAwait(false);
             }
             else
             {
-                throw ExceptionUtilities.UnexpectedValue(declarationContext.Kind());
+                throw ExceptionUtilities.UnexpectedValue(declarationContext?.Kind());
             }
+        }
+        private static async Task HandleDeclarationExpressionAsync(Document document, SyntaxEditor editor, DeclarationExpressionSyntax declarationExpression, CancellationToken cancellationToken)
```

Comment:
```suggestion

        private static async Task HandleDeclarationExpressionAsync(Document document, SyntaxEditor editor, DeclarationExpressionSyntax declarationExpression, CancellationToken cancellationToken)
```

Also, can you doc why this method doesn't need to adjust nullability?